reports-ui/src/main/java/io/jmix/reportsui/screen/parameter/edit/ParameterEditor.java

```diff
@@ -157,15 +157,22 @@
     protected ParameterFieldCreator parameterFieldCreator;
 
     @Subscribe
-    protected void onInit(Screen.InitEvent event) {
+    protected void onInit(InitEvent event) {
         parameterTypeField.setOptionsList(Arrays.asList(ParameterType.TEXT, ParameterType.NUMERIC, ParameterType.BOOLEAN, ParameterType.ENUMERATION,
                 ParameterType.DATE, ParameterType.TIME, ParameterType.DATETIME, ParameterType.ENTITY, ParameterType.ENTITY_LIST));
-
         initMetaClassLookup();
         initEnumsLookup();
         initCodeEditors();
     }
 
+    @Subscribe
+    public void onBeforeShow(BeforeShowEvent event) {
+        if(getEditedEntity().getParameterClass() == null){
```

Comment:
1. Can you extract getEditedEntity() to a variable? 
2. I still don't get what is the problem. Can you add a meaningful comment to this code? Can it affect other editors?

Reply:
1. Ok with this
2. We are invoking this method on every beforeShow event but class property could be already initilized OR report can be imported. By the way better can be OnEntityInit event listener, but we need to extract default value of ParamenerType from screen components. Or pass throught value list, get first value (its string right now) and then set it as constant. Then could be OnEntityInit used. For code capability i prefer null check (is initilized) + beforeShow